flatpack/src/main/java/net/sf/flatpack/AbstractDelimiterParser.java

```diff
@@ -236,101 +236,147 @@ private boolean oddNumberOfQualifier(final String line, final char q) {
      * could span multiple lines.
      * NULL will be returned when the end of the file is reached
      *
-     * @param br
+     * @param aContentReader
      *          Open reader being used to read through the file
-     * @param qual
+     * @param aQualifier
      *          Qualifier being used for parse
-     * @param delim
+     * @param aDelimiter
      *          Delimiter being used for parse
      * @return String
      *          Record from delimited file
      * @throws IOException if any problem with the stream of data (e.g. file reader)
+     *
+     * Improved version of line fetching that solves some of the issues of flatpack parser.
      */
-    protected String fetchNextRecord(final BufferedReader br, final char qual, final char delim) throws IOException {
+    protected String fetchNextRecord(BufferedReader aContentReader, char aQualifier, char aDelimiter) throws IOException

```

Comment:
this is the alternative impl of that should solve some of the problems I found. I tested it as much as I can and saw that it covers the cases I had. But I am not sure if it is well enough to consume in the master as is